### PR TITLE
Improved doxygen for Yarp devices

### PR DESCRIPTION
**Pull request by randaz81**
Improved doxygen documentation for Yarp devices

#2346 needs to be merged first, then rebase

### CODE REVIEW

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+
+bool RGBDSensorFromRosTopic::getImages(FlexImage& colorFrame, ImageOf<PixelFloat>& depthFrame, Stamp* colorStamp, Stamp* depthStamp)
+{
+    bool rgb_ok = false;
+    bool depth_ok = false;
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_rgb_input_processor != nullptr)
+        rgb_ok = m_rgb_input_processor->getLastRGBData(colorFrame, *colorStamp);
+    if (m_depth_input_processor != nullptr)
+        depth_ok = m_depth_input_processor->getLastDepthData(depthFrame, *depthStamp);
```
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:273]: Codacy found an issue: [Variable 'depth_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDRosConversionUtils/rosPixelCode.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    case VOCAB_PIXEL_RGBA:
+        return RGBA8;
+    case VOCAB_PIXEL_MONO_FLOAT:
+        return TYPE_32FC1;
+    default:
+        return RGB8;
+    }
+}
+
+int Ros2YarpPixelCode(std::string roscode)
```
Comment (drdanz) [inline src/devices/RGBDRosConversionUtils/rosPixelCode.cpp:54]: Codacy found an issue: [Function parameter 'roscode' should be passed by const reference.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_depth_input_processor != nullptr)
+        bool depth_ok = m_depth_input_processor->getLastDepthData(depthImage, *timeStamp);
+    return true;
+}
+
+bool RGBDSensorFromRosTopic::getImages(FlexImage& colorFrame, ImageOf<PixelFloat>& depthFrame, Stamp* colorStamp, Stamp* depthStamp)
+{
+    bool rgb_ok = false;
```
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:267]: Codacy found an issue: [The scope of the variable 'rgb_ok' can be reduced.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+    if (m_rgb_input_processor!=nullptr)
+        bool rgb_ok = m_rgb_input_processor->getLastRGBData(rgbImage, *timeStamp);
+    return true;
+}
+
+bool RGBDSensorFromRosTopic::getDepthImage(ImageOf<PixelFloat>& depthImage, Stamp* timeStamp)
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_depth_input_processor != nullptr)
+        bool depth_ok = m_depth_input_processor->getLastDepthData(depthImage, *timeStamp);
```
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:261]: Codacy found an issue: [Variable 'depth_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+    return true;
+}
+
+bool RGBDSensorFromRosTopic::getImages(FlexImage& colorFrame, ImageOf<PixelFloat>& depthFrame, Stamp* colorStamp, Stamp* depthStamp)
+{
+    bool rgb_ok = false;
+    bool depth_ok = false;
```
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:268]: Codacy found an issue: [The scope of the variable 'depth_ok' can be reduced.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:268]: Codacy found an issue: [Variable 'depth_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+{
+    yCWarning(RGBD_ROS_TOPIC) << "getExtrinsicParam not supported";
+    return  false;
+};
+
+bool RGBDSensorFromRosTopic::getRgbImage(FlexImage& rgbImage, Stamp* timeStamp)
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_rgb_input_processor!=nullptr)
+        bool rgb_ok = m_rgb_input_processor->getLastRGBData(rgbImage, *timeStamp);
```
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:253]: Codacy found an issue: [Variable 'rgb_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp (hunk 3 above)
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:267]: Codacy found an issue: [Variable 'rgb_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+    return true;
+}
+
+bool RGBDSensorFromRosTopic::getImages(FlexImage& colorFrame, ImageOf<PixelFloat>& depthFrame, Stamp* colorStamp, Stamp* depthStamp)
+{
+    bool rgb_ok = false;
+    bool depth_ok = false;
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_rgb_input_processor != nullptr)
+        rgb_ok = m_rgb_input_processor->getLastRGBData(colorFrame, *colorStamp);
```
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:271]: Codacy found an issue: [Variable 'rgb_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

**Review by drdanz — CHANGES_REQUESTED** [review]
I think you should rebase this and remove the commits that do not belong to this PR.
Beside that, documentation is always welcome

**Comment by randaz81** [comment]
> I think you should rebase this and remove the commits that do not belong to this PR.
> Beside that, documentation is always welcome

@drdanz As mentioned in the description, to avoid conflicts, _#2346 needs to be merged first, then rebase_, because this PR changes the documentation of that device too.

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp (hunk 3 above)
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:267]: Codacy found an issue: [The scope of the variable 'rgb_ok' can be reduced.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp (hunk 4 above)
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:261]: Codacy found an issue: [Variable 'depth_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp (hunk 5 above)
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:268]: Codacy found an issue: [The scope of the variable 'depth_ok' can be reduced.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp (hunk 3 above)
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:267]: Codacy found an issue: [Variable 'rgb_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDRosConversionUtils/rosPixelCode.cpp (hunk 2 above)
Comment (drdanz) [inline src/devices/RGBDRosConversionUtils/rosPixelCode.cpp:54]: Codacy found an issue: [Function parameter 'roscode' should be passed by const reference.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp (hunk 5 above)
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:268]: Codacy found an issue: [Variable 'depth_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp (hunk 7 above)
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:271]: Codacy found an issue: [Variable 'rgb_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp (hunk 6 above)
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:253]: Codacy found an issue: [Variable 'rgb_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

File: src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp (hunk 1 above)
Comment (drdanz) [inline src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp:273]: Codacy found an issue: [Variable 'depth_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)

**Comment by drdanz** [comment]
@randaz81 I think that if you rebase this, we can merge it

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**